src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala

```diff
@@ -52,6 +60,44 @@ object EMD extends DebugEnhancedLogging {
     }
   }
 
+  def addAgreementFields(emd: EasyMetadata)(implicit s: Settings): Unit = {
+    val agreementPath = new File(s.bagitDir, depositorInfoDir.resolve("agreements.xml").toString)
+    if (agreementPath.exists) {
+      val agreementsXml = XML.loadFile(agreementPath)
+      if (BooleanUtils.toBoolean((agreementsXml \\ "depositAgreementAccepted").text)) {
+        emd.getEmdRights.setAcceptedLicense(true)
+      }
+      addPrivacySensitiveRemark(emd, agreementsXml)
+    }
+    else {
+      logger.info("agreements.xml not found, not setting agreement data")
+    }
+  }
+
+  private def addPrivacySensitiveRemark(emd: EasyMetadata, agreementsXml: Elem): Unit = {
+    val signerId = (agreementsXml \\ "signerId").text
+    Option((agreementsXml \\ "containsPrivacySensitiveData").text) match {
+      case Some(boolText) =>
+        val privacySensitivePart = if (BooleanUtils.toBoolean(boolText)) "DOES"
+                                   else "DOES NOT"
+        emd.getEmdOther.getEasRemarks.add(new BasicRemark(s"Message for the Datamanager: according to the depositor $signerId this dataset $privacySensitivePart contain Privacy Sensitive data."))
+      case None =>
```

Comment:
This can never occur, since that would mean the `agreements.xml` is not valid. This would be caught earlier in `easy-validate-dans-bag`. Hence, you don't need to wrap in an `Option`.

Reply:
according to the xsd: https://easy.dans.knaw.nl/schemas/bag/metadata/agreements/agreements.xsd its a choice element right?
<xs:choice>
<xs:sequence>
<xs:element name="signerId" type="am:SignerType"/>
<xs:element ref="am:dateSigned"/>
<xs:element ref="am:containsPrivacySensitiveData"/>
</xs:sequence>
<xs:element ref="am:notAvailable"/>
</xs:choice>

it is either the sequence or not available?